### PR TITLE
Pydantic 2 compatibility using `pydantic.v1` 

### LINKED ISSUE #6350: Compatibility with pydantic versions >= 2
## 🚀 Feature

This is a feature request is for compatibility with pydantic v2. 
Just adding an issue because I couldn't find one that tracks it. 

## Motivation

I know that  @tlambert03  and @jni [did already discuss this problem with the pydantic people before v2 was released](https://github.com/pydantic/pydantic/discussions/5402):

Napari cannot currently be installed in a python environment where pydantic >= 2. 
Previously, a suggested work-around was pinning pydantic <2 , see e.g. #6029
As more and more other libraries will depend on pydantic > 2, this is not viable long-term (in fact, it is causing problems now).

## Additional context

* @tlambert03 has been working on a compatibility wrapper https://github.com/tlambert03/pydantic-compat
* Pydantic v2 includes a whole pydantic v1 that can be imported like `from pydantic.v1 import BaseModel`. 
* `app-model` also has some compatibility layer (https://github.com/pyapp-kit/app-model/blob/main/src/app_model/_pydantic_compat.py) which I guess is an early version of   https://github.com/tlambert03/pydantic-compat
* `npe2` also requires pydantic < 2

I tried the naive approach of replacing all imports from pydantic with imports of pydantic.v1 in napari and npe2. (`app-kit` is a different story as it is trying to guess which version is used ... for testing purposes I patched it to import from `pydantic.v1` but set `PYDANTIC2` to `False` despite using pydantic 2.4). Well, I did not get very far, various pydantic related issues popped up that I didn't have time to investigate or document further. I have a feeling this is a hard problem and even if it can be fixed for napari it will break some plugins. (as mentioned in the discussion linked to above). 

It is also not a problem that can be fixed with a single PR to napari, but requires corresponding PRs in npe2 and probably app-model.

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

closes #6350

# Description

Make napari ready on pydantic 2. However, most tests are currently executed on pydantic 1

### CODE REVIEW

File: .github/workflows/test_pull_requests.yml [hunk 1]
```diff
@@ -107,6 +111,11 @@ jobs:
             backend: pyqt5
             MIN_REQ: 1
             coverage: cov
+          - python: "3.10"
+            platform: ubuntu-22.04
+            backend: pyqt5
+            coverage: cov
+            pydantic: ""
           # test without any Qt backends
```
Comment (brisvag) [inline .github/workflows/test_pull_requests.yml:119]: Not sure I follow the changes to the workflows. So there's one extra env in the matrix which tests pydantic>=2 with python 3.10, and the rest is same as before?

Comment (Czaki) [reply .github/workflows/test_pull_requests.yml:119]: One of the initial checks is also pydantic 2. I do not have an opinion how to do this best.

Comment (brisvag) [reply .github/workflows/test_pull_requests.yml:119]: Got it!

**Review by brisvag — APPROVED** [review]
IMO we should get this in, regardless of if the pyapp-kit discussion results in a better solution; this is easy to replace, and for now it's holding up v0.4.19.

**Review by jni — APPROVED** [review]

**Comment by Czaki** [comment]
@napari-bot update constraints

**Pull request merged**